draft-ietf-webtrans-http2.md

```diff
@@ -251,368 +278,411 @@ unidirectional, indicated by the second least significant bit of the
-   streams allowed at the time the frame was sent. This value cannot exceed
-   2<sup>60</sup>, as it is not possible to encode stream IDs larger than 2<sup>62</sup>-1.
+   : A variable-length integer indicating the maximum number of streams allowed
+     at the time the capsule was sent. This value cannot exceed 2<sup>60</sup>,
+     as it is not possible to encode stream IDs larger than 2<sup>62</sup>-1.
 
-## WT_DATAGRAM Frames {#WT_DATAGRAM}
+The WT_STREAMS_BLOCKED capsule defines special intermediary handling, as
+described in {{Section 3.2 of HTTP-DATAGRAM}}.  Intermedaries MUST consume
+WT_STREAMS_BLOCKED capsules for flow control purposes and MUST generate and
+send approrpiate flow control signals for their limits; see
+{{flow-control-intermediaries}}.
 
-*[WT_DATAGRAM]: #
+# DATAGRAM Capsule {#DATAGRAM_CAPSULE}
 
-The WT_DATAGRAM frame type (0x31) is used to carry datagram traffic. Frame type
-0x30 is also reserved to maintain parity with QUIC, but unused, as all
-WebTransport frames MUST contain a length field.
+*[DATAGRAM_CAPSULE]: #
+
+WebTransport over HTTP/2 uses the DATAGRAM capsule defined in {{Section 3.5 of
+HTTP-DATAGRAM}} to carry datagram traffic.
 
 ~~~
-WT_DATAGRAM Frame {
-  Type (i) = 0x31,
+DATAGRAM Capsule {
+  Type (i) = 0x00,
   Length (i),
-  Datagram Data (..),
+  HTTP Datagram Payload (..),
 }
 ~~~
-{: #fig-wt_datagram title="WT_DATAGRAM Frame Format"}
+{: #fig-datagram title="DATAGRAM Capsule Format"}
 
-WT_DATAGRAM frames contain the following fields:
+When used in WebTransport over HTTP/2, DATAGRAM capsules contain the following
+fields:
 
-   Datagram Data:
+   HTTP Datagram Payload:
    : The content of the datagram to be delivered.
 
-The data in WT_DATAGRAM frames is not subject to flow control. The receiver MAY
+The data in DATAGRAM capsules is not subject to flow control. The receiver MAY
 discard this data if it does not have sufficient space to buffer it.
 
-An intermediary could forward the data in a WT_DATAGRAM frame over another
-protocol, such as WebTransport over HTTP/3. In QUIC, a datagram frame can span
+An intermediary could forward the data in a DATAGRAM capsule over another
+protocol, such as WebTransport over HTTP/3.  In QUIC, a datagram frame can span
 at most one packet. Because of that, the applications have to know the maximum
 size of the datagram they can send. However, when proxying the datagrams, the
 hop-by-hop MTUs can vary.
 
+{{Section 3.5 of HTTP-DATAGRAM}} indicates that intermediaries that forward
+DATAGRAM capsules where QUIC datagrams {{DATAGRAM}} are available forward the
+contents of the capsule as native QUIC datagrams, rather than as HTTP datagrams
+in a DATAGRAM capsule. Similarly, when forwarding DATAGRAM capsules used as
+part of a WebTransport over HTTP/2 session on a WebTransport session that
+natively supports QUIC datagrams, such as WebTransport over HTTP/3
+{{WEBTRANSPORT-H3}}, intermediaries SHOULD use native QUIC datagrams.
```

Comment:
We should not levy requirements on HTTP/3 webtransport.  An informative reference is fine, but avoid 2119 language.

Doubly so because I believe we agreed to a MUST here.

Reply:
Excellent point, we should have some text somewhere about forwarding between the two, but worst(/best?) case we can have each document describe it's normative requirements and then just reference the other.

Reply:
Fixed to just say "do the thing H3 says" (which is MUST be datagrams) with a reference to go look over there.